src/realm/object-store/sync/app.cpp

```diff
@@ -174,7 +175,8 @@ const static uint64_t default_timeout_ms = 60000;
 const static std::string username_password_provider_key = "local-userpass";
 const static std::string user_api_key_provider_key_path = "api_keys";
 const static int max_http_redirects = 20;
-static std::unordered_map<std::string, std::shared_ptr<App>> s_apps_cache;
+static std::unordered_map<std::string, std::unordered_map<std::string, SharedApp>>
```

Comment:
considering that there will be only one of these in most cases and at most there will be 2 or 3, we can use a `FlatMap` from `realm/util/flat_map.hpp` here and skip the overhead of a hash map.